### PR TITLE
chore: Added Store

### PR DESCRIPTION
**Pull request by sawood14012**
- Added store
- Added a small version API call
- Converted Dashboard to FC

### CODE REVIEW

File: frontend/src/app/store/reducer.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    switch (action.type) {
+        case 'APIData':
+            return {
+                ...state,
+                APIData: action.data
+            };
+        case 'SET_Version':
+            return {
+                ...state,
+                vesrion: action.data
```
Comment (flacatus) [inline frontend/src/app/store/reducer.tsx]: version?

Comment (sawood14012) [reply frontend/src/app/store/reducer.tsx]: "dashboard version" i just added it for better understanding of global state

Comment (flacatus) [reply frontend/src/app/store/reducer.tsx]: I mean: vesrion should be version

Comment (sawood14012) [reply frontend/src/app/store/reducer.tsx]: oh sorry a typo

Comment (sawood14012) [reply frontend/src/app/store/reducer.tsx]: fixed it

**Pull request merged**